tower-http/src/trace/on_request.rs

```diff
@@ -1,26 +1,30 @@
 use super::DEFAULT_MESSAGE_LEVEL;
 use http::Request;
 use tracing::Level;
+use tracing::Span;
 
 /// Trait used to tell [`Trace`] what to do when a request is received.
 ///
 /// [`Trace`]: super::Trace
 pub trait OnRequest<B> {
     /// Do the thing.
-    fn on_request(&mut self, request: &Request<B>);
+    ///
+    /// `span` can be used to record field values that weren't known when the span was
+    /// created.
```

Comment:
and again, maybe
```suggestion
    /// `span` is the `tracing` [`Span`] corresponding to this request, produced
    /// the closure passed to [`TraceLayer::make_span_with`]. It can be used to
    /// [record field values][record] that weren't known when the span was
    /// created.
    ///
    /// [`Span`]: https://docs.rs/tracing/latest/tracing/span/index.html
    /// [record]: https://docs.rs/tracing/latest/tracing/span/struct.Span.html#method.record
```